packages/key-readmodel-writer-sql/src/keyConsumerServiceV2.ts

```diff
@@ -0,0 +1,69 @@
+import { keyToClientJWKKey } from "pagopa-interop-commons";
+import {
+  AuthorizationEventEnvelopeV2,
+  fromClientV2,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import { CustomReadModelService } from "./readModelService.js";
+
+export async function handleMessageV2(
+  message: AuthorizationEventEnvelopeV2,
+  clientJWKKeyReadModelService: CustomReadModelService
+): Promise<void> {
+  await match(message)
+    .with({ type: "ClientKeyAdded" }, async (message) => {
+      const client = message.data.client
+        ? fromClientV2(message.data.client)
+        : undefined;
+      if (!client) {
+        throw Error("Client not found in event");
```

Comment:
Usually, we try to not throw an `Error` directly but to use a function that we define and control, in these cases I think we use `genericInternalError` (same for all other throws in the PR)

```suggestion
        throw genericInternalError("Client not found in event");
```